app/assets/javascripts/listings/ListingDataService.js.coffee

```diff
@@ -0,0 +1,434 @@
+      return
+    )
+
+  Service._consolidatedAMICharts = (amiData) ->
+    charts = []
+    amiData.forEach (chart) ->
+      # look for an existing chart at the same percentage level
+      amiPercentChart = _.find charts, (c) -> c.percent == chart.percent
+      if !amiPercentChart
+        # only push chart if it has any values
+        charts.push(chart) if chart.values.length
+      else
+        # if it exists, modify it with the max values
+        i = 0
+        amiPercentChart.values.forEach (incomeLevel) ->
+          chartAmount = if chart.values[i] then chart.values[i].amount else 0
+          incomeLevel.amount = Math.max(incomeLevel.amount, chartAmount)
+          i++
+    charts
+
+  Service.priorityTypes = (listing) ->
+    Service.collectTypes(listing, 'prioritiesDescriptor')
+
+  Service.collectTypes = (listing, specialType) ->
+    _.map listing[specialType], (descriptor) ->
+      descriptor.name
+
+  Service.priorityLabel = (priority, modifier) ->
+    labelMap =
+      'Vision impairments':
+        name: 'Vision Impairments'
+        description: 'impaired vision'
+      'Hearing impairments':
+        name: 'Hearing Impairments'
+        description: 'impaired hearing'
+      'Hearing/Vision impairments':
+        name: 'Vision and/or Hearing Impairments'
+        description: 'impaired vision and/or hearing'
+      'Mobility/hearing/vision impairments':
+        name: 'Mobility, Hearing and/or Vision Impairments'
+        description: 'impaired mobility, hearing and/or vision'
+      'Mobility impairments':
+        name: 'Mobility Impairments'
+        description: 'impaired mobility'
+
+    return priority unless labelMap[priority]
+    return labelMap[priority][modifier]
+
+  Service.reservedLabel = (listing, type,  modifier) ->
+    labelMap =
+      "#{ListingConstantsService.RESERVED_TYPES.SENIOR}":
+        building: 'Senior'
+        eligibility: 'Seniors'
+        reservedFor: "seniors #{Service.seniorMinimumAge(listing)}"
+        reservedForWhoAre: "seniors #{Service.seniorMinimumAge(listing)}"
+        unitDescription: "seniors #{Service.seniorMinimumAge(listing)}"
+      "#{ListingConstantsService.RESERVED_TYPES.VETERAN}":
+        building: 'Veterans'
+        eligibility: 'Veterans'
+        reservedFor: 'veterans'
+        reservedForWhoAre: 'veterans'
+        unitDescription: 'veterans of the U.S. Armed Forces'
+      "#{ListingConstantsService.RESERVED_TYPES.DISABLED}":
+        building: 'Developmental Disability'
+        eligibility: 'People with developmental disabilities'
+        reservedFor: 'people with developmental disabilities'
+        reservedForWhoAre: 'developmentally disabled'
+        unitDescription: 'people with developmental disabilities'
+
+    return type unless labelMap[type]
+    return labelMap[type][modifier]
+
+  # used by My Applications -- when you load an application we also parse the attached listing data
+  Service.loadListing = (listing) ->
```

Comment:
I'm not really clear on what's going on in this function. if the listing passed matches the listing we already have, return, else set the listing we have (from somewhere else) into the listingDataService?